lib/auth/v4/validateInputs.js

```diff
@@ -109,6 +109,12 @@ function extractQueryParams(queryObj, log) {
     const credential = queryObj['X-Amz-Credential'];
     if (credential && credential.length > 28 && credential.indexOf('/') > -1) {
         authParams.credential = credential.split('/');
+        // in case access keys contains a '/', reduce the credentials array to
+        // the desire size by recomputing the access key
+        while (authParams.credential.length > 5) {
```

Comment:
So we're assuming that in this "credential", where AccessKey is supposed to be first; we concatenate all the early items in the array, until we only have 5 left ?

ie, assuming an array length of X after splitting (python example for ease of explanation):
```py
'/'.join(creds[0:X-4])
```

Does this mean that no other component can have a '/' within ? If so, it might deserve an additional comment to explain the assumptions made in this code, and validate its correctness.

Reply:
yes, not other component will have a '/' :)

Reply:
okay, then a comment could be nice. I'm also guessing there's no specification of the AccessKeyLength, that could help double-checking that we're not messing with the creds ?

Reply:
the length can change, but the signature computation is secure enought :)